### DISCUSSION TITLE
Check file truncate by signature是否会引起日志收集中断或者丢失

### DISCUSSION THREAD

**Discussion by hailangxing** [alibaba/ilogtail#1347] [Help] [answered]
![96708DE0-A2F8-4c53-9EB7-089E824A721D](https://github.com/alibaba/ilogtail/assets/8348166/3b64d34c-a3cf-4f54-bc2c-ee62162b913d)

**Comment by 7y-9** [comment]
\t[]\t[3781474]\t/src/core/reader/LogFileReader.cpp:1372\t\tFile signature is same but size decrease, read from now fileSize:/data/logs/***/***/system/info.log\t0:41624096670\tproject:\tlogstore:\tconfig:config#/****//user_yaml_config.d/5749_3a074d_bdd672.yaml

我这边也遇到了这种，触发了 从头采集

**Comment by yyuuttaaoo** [comment, marked as answer]
这个日志的意思是文件的签名变了，说明日志文件和原来记录的不是同一个了。看后一行日志，看起来像是inode复用，会重建reader读取新文件。这样并不会导致数据重复。

**Comment by 7y-9** [comment]
@yyuuttaaoo 感谢回复，我们遇到这个从头采集是这样的场景。
运维有一个定期执行的脚本 cp /dev/null 到日志文件，执行之后，ilogtail会报这个错，会从头进行采集。
当文件fd被进程持有，请问有什么推荐清理日志文件的方式吗
